frontend/packages/ceph-storage-plugin/src/components/modals/clone-pvc-modal/clone-pvc-modal.tsx

```diff
@@ -0,0 +1,193 @@
+import { PersistentVolumeClaimModel } from '@console/internal/models/index';
+import {
+  DashboardItemProps,
+  withDashboardResources,
+} from '@console/internal/components/dashboard/with-dashboard-resources';
+import { HandlePromiseProps } from '@console/internal/components/utils/promise-component';
+import { PrometheusRulesResponse } from '@console/internal/components/monitoring';
+import { getPVCUsedCapacityQuery } from '../../../constants/queries';
+
+import './_clone-pvc-modal.scss';
+
+const accessModeRadios = {
+  ReadWriteOnce: 'Single User (RWO)',
+  ReadWriteMany: 'Shared Access (RWX)',
+  ReadOnlyMany: 'Read Only (ROX)',
+};
+
+export const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
+  const {
+    close,
+    cancel,
+    resource,
+    watchPrometheus,
+    stopWatchPrometheusQuery,
+    prometheusResults,
+    handlePromise,
+  } = props;
+  const pvcName: string = _.get(resource, 'metadata.name');
+  const [buttonDisabled, setButton] = React.useState(false);
+  const [inProgress, setProgress] = React.useState(false);
+  const [errorMessage, setError] = React.useState('');
+  const [clonePVCName, setClonePVCName] = React.useState(`${pvcName}-clone`);
+
+  const query: string = getPVCUsedCapacityQuery(pvcName);
+
+  React.useEffect(() => {
+    watchPrometheus(query);
+    return () => stopWatchPrometheusQuery(query);
+  }, [watchPrometheus, stopWatchPrometheusQuery, query]);
+
+  const results = prometheusResults.getIn([query, 'data']) as PrometheusRulesResponse;
+  const pvcUsedCapacity: string = _.get(results, 'data.result[0].value[1]');
+  const queriesLoadError = prometheusResults.getIn([query, 'loadError']);
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    setProgress(true);
+
+    const pvcCloneObj = {
+      apiVersion: 'v1',
+      kind: PersistentVolumeClaimModel.kind,
+      metadata: {
+        name: clonePVCName,
+        namespace: _.get(resource, 'metadata.namespace'),
+      },
+      spec: {
+        storageClassName: _.get(resource, 'spec.storageClassName'),
+        dataSource: {
+          name: pvcName,
+          kind: 'PersistentVolumeClaim',
+          apiGroup: '',
+        },
+        accessModes: _.get(resource, 'spec.accessModes'),
+        resources: {
+          requests: {
+            storage: _.get(resource, 'spec.resources.requests.storage'),
+          },
+        },
+      },
+    };
+
+    handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj))
+      .then(() => {
+        setProgress(false);
+        close();
+      })
+      .catch((error) => {
+        setError(error);
+        setProgress(false);
+        setButton(true);
+        throw error;
+      });
+  };
+
+  const updateClonePVCName = (value: string) => {
+    setClonePVCName(value);
+  };
+
+  return (
+    <Form onSubmit={submit}>
+      <div className="modal-content modal-content--no-inner-scroll">
+        <ModalTitle>Clone</ModalTitle>
```

Comment:
Wont this be "Clone PVC" in alignment with "Expand PVC" ?

Reply:
No, as per UXD, its `Clone` only. https://marvelapp.com/100ah4f4/screen/64186762